packages/playwright/src/isomorphic/testServerConnection.ts

```diff
@@ -19,6 +19,16 @@ import * as events from './events';
 
 // -- Reuse boundary -- Everything below this line is reused in the vscode extension.
 
+export interface TestServerSocket {
```

Comment:
- We usually call things like this one "Transport" for easier search across the codebase.
- This is a somewhat convoluted interface for a transport. Perhaps we should separate transport (onmessage, onclose, send, close) from a factory that creates a websocket transport (this one needs open/error/etc).

Reply:
Sounds good! How do you like https://github.com/microsoft/playwright/pull/32274/commits/0f37e55eeca8b81e5bc524088dafd213db98e8fc?